pylint/checkers/utils.py

```diff
@@ -402,7 +402,10 @@ def is_default_argument(
     if not scope:
         scope = node.scope()
     if isinstance(scope, (nodes.FunctionDef, nodes.Lambda)):
-        for default_node in scope.args.defaults:
+        all_defaults = scope.args.defaults + [
```

Comment:
```python
    return any(
        default_name_node is node
        for default_name_node in default_node.nodes_of_class(nodes.Name)
        for default_node in scope.args.defaults +  d for d in scope.args.kw_defaults if d is not None
    )
```
Maybe something like this would work ? I'm a little rusty with the imbricated for loop.

Reply:
Nice catch, yeah I was being lazy before. You can't directly concatenate a list and a generator, but I used `itertools.chain` to the same effect.